### ISSUE TITLE
Release automation pipeline

### ISSUE THREAD

**Issue by vasilich6107** [joanpablo/reactive_forms#198]
Hi.
Could you setup automation for releasing the reactive forms version after merging PR?

**Comment by joanpablo** [comment]
Hi @vasilich6107,

Could you explain to me a little better how to set up automation for release?

Thanks

**Comment by vasilich6107** [comment]
ok I will write my thoughts

**Comment by kuhnroyal** [comment]
I don't think releases after every PR are the way to go.

**Comment by vasilich6107** [comment]
@kuhnroyal could you add any cons?

**Comment by kuhnroyal** [comment]
Releases should be a conscious decisions. Some PRs are not worth spamming releases, some may just do some refactoring. Releases require decisions about what is a breaking change and what is not, what should the next version number be. You can't expect this from every PR in an OS project.

It would be possible to make `develop` the default branch and merge all PRs into this one.
Then you could add release validation and automation for PRs from `develop` to `master`.

**Comment by vasilich6107** [comment]
@kuhnroyal you are definitely right
what if we could setup immediate releases into `beta` from `develop` and `stable` releases from master?

I have an experience helping to maintain `artemis` package. We have `beta` branch which is instantly delivered as `beta` version of the package. The aim is to deliver releases as fast as possible to the user.

**Comment by joanpablo** [comment]
I totally agree with @kuhnroyal. I want to carefully decide what to release as an official version of the package. I also don't want that the version number of the package grow so quickly (It already has grown fast)

But if you have some ideas, please you're welcome to detail what you have in mind.

**Comment by vasilich6107** [comment]
Hi)
@joanpablo could you consider adding me and @kuhnroyal as maintainers and organize some kind of automation pipeline at least for beta to facilitate library evolution?

**Comment by vasilich6107** [comment]
@joanpablo would be great to hear you thoughts

**Comment by vasilich6107** [comment]
@joanpablo could you take some time to discuss this issue?

**Comment by mdeandrea-mrmilu** [comment]
Take a look [on this](https://github.com/flutter-form-builder-ecosystem/flutter_form_builder/blob/0b8f8e96d03213501d161dc7582ba2c0bb659350/.github/workflows/base.yaml#L58). 
Maybe can implement in this repo

**Comment by babaralishah** [comment]
Hi 
I am on Angular

And using the below regex:

`'^[a-z0-9]+([._-]?[a-z0-9]+)+@[a-z0-9]+([._-]?[a-z0-9]+)+\\.[a-z]{2,3}$'`

But its not fulfilling my requirement, my requirement is below:

`saa5@me-d_d_u.co`

please anyone tell me, why this regex is not working for my above mentioned case.

Regards
Babar Ali Shah

**Comment by joanpablo** [comment]
Hi @vasilich6107 and @kuhnroyal I'm more that glad to implement some kind of flow in whish you both are involved, and to accept PR based on scores for example a PR must have more than 2 approvals and something like that. I have been quiet busy in las 4 months. I'm agree with guys. I still need to take a look at the PR and the link that @mdeandrea-mrmilu posted. But if you have any idea about how to discuss the flow, present examples, doc anything is very welcome
